From 050a08c440e2cd1e475d0913a124c700d3c67f99 Mon Sep 17 00:00:00 2001
From: Wenju He <wenju.he@intel.com>
Date: Mon, 22 May 2023 13:10:03 +0800
Subject: [PATCH 1004/1004] Fix OPENCL_HEADERS_DIR out-of-tree build (#442)

On ubuntu 20.04 of my local machine, clang install folder is
/usr/lib/llvm-16/lib/clang/16/include
---
 cl_headers/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/cl_headers/CMakeLists.txt b/cl_headers/CMakeLists.txt
index 86df76f..0702d89 100644
--- a/cl_headers/CMakeLists.txt
+++ b/cl_headers/CMakeLists.txt
@@ -19,8 +19,15 @@ add_custom_command(
 endfunction(copy_file)
 
 if(USE_PREBUILT_LLVM)
-    set(OPENCL_HEADERS_DIR
+  set(OPENCL_HEADERS_DIR
       "${LLVM_LIBRARY_DIRS}/clang/${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}/include/")
+  if(NOT EXISTS ${OPENCL_HEADERS_DIR})
+    set(OPENCL_HEADERS_DIR
+        "${LLVM_LIBRARY_DIRS}/clang/${LLVM_VERSION_MAJOR}/include/")
+  endif()
+  if(NOT EXISTS ${OPENCL_HEADERS_DIR})
+    message(FATAL_ERROR "[OPENCL-CLANG] Unable to find clang include folder at ${OPENCL_HEADERS_DIR}")
+  endif()
 else(USE_PREBUILT_LLVM)
     set(OPENCL_HEADERS_DIR "${CLANG_SOURCE_DIR}/lib/Headers")
 endif(USE_PREBUILT_LLVM)
-- 
2.40.1

